packages/malloy/src/model/malloy_types.ts

```diff
@@ -38,15 +38,58 @@ export function paramHasValue(p: Parameter): boolean {
 }
 
 export interface DocumentRange {
-  start: { line: number; character: number };
-  end: { line: number; character: number };
+  start: DocumentPosition;
+  end: DocumentPosition;
+}
+
+export interface DocumentPosition {
+  line: number;
+  character: number;
 }
 
 export interface DocumentLocation {
   url: string;
   range: DocumentRange;
 }
 
+interface DocumentReferenceBase {
+  text: string;
+  location: DocumentLocation;
+  definition: HasLocation;
+}
+
+export interface DocumentExploreReference extends DocumentReferenceBase {
```

Comment:
should these be `stuctDef: StructDef`, `fieldDef: FieldEf` instead of `definition: ` ?

Reply:
I see your point... though it's just nice to always be able to say `reference.definition.location`.